lib/rack/utils.rb

```diff
@@ -251,7 +251,11 @@ def add_cookie_to_header(header, key, value)
         same_site = if value[:same_site]
           case value[:same_site]
           when Symbol, String
-            "; SameSite=#{value[:same_site]}"
+            same_site_rule = value[:same_site].to_s
+            unless ["Lax", "Strict"].include?(same_site_rule)
+              raise ArgumentError, "Unrecognized cookie header value for SameSite option: #{same_site_rule}"
+            end
+            "; SameSite=#{same_site_rule}"
           else
             "; SameSite"
           end
```

Comment:
We can dial this back to a stricter whitelist now that the spec requires valid values, e.g.
```ruby
same_site =
  case value[:same_site]
  when false, nil
    nil
  when :lax, 'Lax'
    '; SameSite=Lax'.freeze
  when true, :strict, 'Strict'
    '; SameSite=Strict'.freeze
  else
    raise ArgumentError, "Invalid SameSite value: #{value[:same_site].inspect}"
  end
```